test/tests/functional/pbs_hook_execjob_prologue.py

```diff
@@ -249,10 +249,13 @@ def test_prologue_hook_does_not_execute_twice_on_pbsdsh(self):
 
         j = Job(TEST_USER, {'Resource_List.select': '2:ncpus=1',
                             'Resource_List.place': 'scatter'})
-        j.create_script('pbsdsh -n 1 hostname')
+        j.set_sleep_time("5")
```

Comment:
however I have learnt that it is not mandatory to have pbsdsh for exec job_prologue hook now to run prologue hook on sister moms, but this test is about running pbsdsh.

Your changes are fine but not testing what this test was supposed to do. instead of reading from exec_vnode read from the job output file. update the job script to have "pbsdsh hostname" and submit a 2 chunk job. Read the output file and get the hostnames on which this job has run. for example,

anamika@testdev-01-s11p2:~> cat job1.sh.o9
x40-64-centos64.pbspro.com
testdev-01-s11p2

and then look for the "Job;9.testdev-01-s11p2;executed prologue hook" in mom_logs of both the hosts.

Reply:
I've made the changes as you suggested. However, I've removed the "-n 1" from the job script as it was causing only the mother superior to be printed in the job o/p file.